crates/interfaces/src/p2p/headers/client.rs

```diff
@@ -23,12 +23,24 @@ pub struct HeadersRequest {
 /// The headers future type
 pub type HeadersFut = Pin<Box<dyn Future<Output = PeerRequestResult<Vec<Header>>> + Send + Sync>>;
 
+/// A type representing either a valid header or invalid header
+#[derive(Debug, Clone)]
+pub enum ValidatedHeader {
+    /// A valid header
+    Valid(Header),
+    /// An invalid header
+    Invalid(Header),
+}
+
 /// The block headers downloader client
 #[auto_impl::auto_impl(&, Arc, Box)]
 pub trait HeadersClient: DownloadClient {
     /// The headers future type
     type Output: Future<Output = PeerRequestResult<Vec<Header>>> + Sync + Send + Unpin;
 
+    /// The future type for a single validated header
+    type ValidatedOutput: Future<Output = PeerRequestResult<ValidatedHeader>> + Sync + Send + Unpin;
+
```

Comment:
do we actually need this here in the download?

I'm a bit skeptical that there's a benefit, because this now requires that we add a chainspec to all downloaders
and validated is also ambiguous because we can only really do standalone validation here.

imo all of this should be handled by the type that's using the downloader, the full block downloader is a special case though